libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFAction.m

```diff
@@ -131,16 +129,19 @@ + (NSError *)errorInResponseDataForAction:(CSFAction*)action {
 
 + (instancetype)actionWithHTTPMethod:(NSString*)method onURL:(NSURL*)url withResponseBlock:(CSFActionResponseBlock)responseBlock {
     CSFAction *action = [[self alloc] initWithResponseBlock:responseBlock];
-    
     NSString *baseString = nil;
     if (url.port) {
         baseString = [NSString stringWithFormat:@"%@://%@:%@", url.scheme, url.host, url.port];
     } else {
         baseString = [NSString stringWithFormat:@"%@://%@", url.scheme, url.host];
     }
-    
     action.baseURL = [NSURL URLWithString:baseString];
-    action.verb = url.path;
+    NSMutableString *relativePath = [NSMutableString stringWithString:url.path];
```

Comment:
This bug is somewhat interesting. We read the ```path``` here from the URL, but completely ignore any query params that may be set on the URL itself. This snippet of code adds the query params to ```verb``` too, which in turn adds them as ```CSFParameterStorage```. That's the good news. The bad news is that ```action.params``` is ignored as well in ```urlForAction```. I attempted to fix that, but it's impossible without some serious changes to the way that stuff is stored/handled/architected. That's a battle for another day.

Moral of the story: This class wasn't really built for external URLs @khawkins.